wagtail/admin/views/pages/listing.py

```diff
@@ -20,13 +28,33 @@
     PageTable,
     PageTitleColumn,
 )
-from wagtail.admin.views.generic.base import BaseListingView
-from wagtail.admin.views.generic.permissions import PermissionCheckedMixin
-from wagtail.models import Page
+from wagtail.admin.views.generic.models import IndexView as GenericIndexView
+from wagtail.models import Page, get_page_models
 from wagtail.permissions import page_permission_policy
 
 
-class BaseIndexView(PermissionCheckedMixin, BaseListingView):
+def get_content_types_for_filter():
+    models = [model.__name__.lower() for model in get_page_models()]
+    return ContentType.objects.filter(model__in=models).order_by("model")
+
+
+class PageFilterSet(WagtailFilterSet):
```

Comment:
Unfortunately the `search_all` parameter isn't retained upon applying the filters.

I managed to hack it to work by adding the following:

```suggestion
from django.forms import HiddenInput

class PageFilterSet(WagtailFilterSet):
    def noop_filter(self, qs, name, value):
        """
        A filter method that does not affect the queryset, useful for retaining
        query parameters that are handled outside of the filterset.
        """
        return qs

    search_all = django_filters.BooleanFilter(widget=HiddenInput, method="noop_filter")
```

If we want this, I think the `noop_filter` method would be a useful addition to `WagtailFilterSet`.

Reply:
Making the filter form responsible for keeping track of the `search_all` flag feels a bit iffy - that's a detail relating to the `<form>` element that contains both the search and filter forms, and I feel like we've been tying ourselves in knots by not being rigorous enough about deciding who is responsible for that form, what it contains, and under what circumstances it gets submitted. Alongside this particular bug, this is manifesting itself in various other places:

* Searching _does_ respect search_all, because it's not actually submitting/serialising the form - it's plucking out the search input value and combining that with the existing URL querystring, which means that searches don't reflect any changes to filters that have just been made but not submitted
* the need to scatter `is_searching or is_filtering` conditions through the table rendering - it would be neater to treat filtering as a kind of search, but there's code elsewhere in the view that interprets `is_searching` as "send this to a search backend"

I notice that the designs don't have any explicit submit buttons, which might imply that filters are intended to take effect immediately on click/edit, much like search does - this would probably go some way to tightening up this fuzziness. Will ponder some more...

Reply:
I noticed that the SwapController code already serializes the form **if you use the `submit` method** (not the `search` method), so using `submitLazy` instead of `searchLazy` in the `data-action`, together with my hack above, search+filtering will work with AJAX and `search_all` is retained.

Reply:
Although that does not clear the `p` param – can probably just copy-paste that from the `search` method. At which point I'm not sure if we'll still have any use for the `search` method instead of the `submit`, provided we use the controller with `SearchForm`...